### PR TITLE
Add comment to unused argument: sample_weight_mode

### PR DESCRIPTION
**Pull request by howl-anderson**
Argument: `sample_weight_mode` of Keras Model:compile is no longer needed but for backward compatibility, we need to keep it untouched.

This PR adds a comment for it.

related to #37125

### CODE REVIEW

File: tensorflow/python/keras/engine/training.py [hunk 1]
```diff
@@ -306,6 +306,9 @@ def compile(self,
         ValueError: In case of invalid arguments for
             `optimizer`, `loss`, `metrics` or `sample_weight_mode`.
     """
+    # NOTE: Argument `sample_weight_mode` is not used,
```
Comment (omalleyt12) [inline tensorflow/python/keras/engine/training.py]: Can we move this into the docstring for `sample_weight_mode`? I.e.

```
Arguments:
  ...
  sample_weight_mode: Unused argument. Note that passing this argument is no longer necessary. In `Model.fit`, 
   `sample_weight` can be either a 1-d Tensor for per-sample weighting, or a 2-d Tensor for per-sample-and-timestep 
       weighting.
```

**Review by omalleyt12 — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: tensorflow/python/keras/engine/training.py (hunk 1 above)
Comment (howl-anderson) [reply tensorflow/python/keras/engine/training.py]: Good idea!

**Comment by howl-anderson** [comment]
@omalleyt12 I updated the PR, please take a look. Thank you!

**Comment by gbaned** [comment]
@howl-anderson Can you please resolve conflicts? Thanks!

**Comment by howl-anderson** [comment]
@gbaned Sure, I submit a new commit for this later.

**Comment by howl-anderson** [comment]
@gbaned conflicts resolved.

**Comment by howl-anderson** [comment]
@omalleyt12 Please take a look. Thank you!

**Comment by gbaned** [comment]
@howl-anderson Can you please resolve conflicts? Thanks!

**Comment by howl-anderson** [comment]
Hi @gbaned and @omalleyt12, recently, a commit [Remove sample_weight_mode arg from compile and cleanup docs](https://github.com/tensorflow/tensorflow/commit/c7aa3e8d1711f49add2c6dc76fc8c000ce1edee8) merged into the master branch by @pavithrasv, that commit fixes the issue this PR attend to solve. So this PR will be closed since it already finishes its purpose. Thank you all for efforts!

**Pull request closed without merging**